### ISSUE TITLE
Using WSS with localhost

### ISSUE THREAD

**Issue by jpaulm** [jpaulm/javafbp-websockets#35]
I would like to expand my code to let the JavaScript 'chat' scripts use 'wss://localhost' - I believe I need to create a certificate and put it in Trusted Root Cetification Authorities.  I would greatly appreciate it if anyone who knows about this stuff could give me a hand!

**Comment by jpaulm** [comment]
I have basically given up on working with certificates, and am testing using Chrome on the client side, with the option `chrome://flags/#allow-insecure-localhost` specified...

IIUC the Java-WebSocket project on GitHub, which this is based on, no longer supports WS - if true, not sure why this change was made.   

Again, help would be much appreciated!

**Comment by jpaulm** [comment]
IIUC the Java-WebSocket project on GitHub, which this is based on, no longer supports WS - if true, not sure why this change was made.   I am using WSS, but have basically given up on working with certificates, so I am testing using Chrome on the client side, with the option `chrome://flags/#allow-insecure-localhost` specified...

Again, help would be much appreciated!  @tootallnate @marci4...

**Comment by marci4** [comment]
No this is not correct, we did not drop WS support.

**Comment by jpaulm** [comment]
@marci4 That's great news!  I will (try to!) reinstate the WS support in my software...  As a parallel effort, can you point me at a *simple* explanation of how to create a trusted certificate...   I described what worked briefly for me in https://github.com/jpaulm/javafbp-websockets/blob/master/README.md - but it stopped working a few days later IIRC - and I wasn't able to get it working again!  Really sorry to bother you!  TIA

**Comment by marci4** [comment]
You cannot get a trusted certificate for localhost/127.0.0.1 to my knowledge.
If you are in control of a domain, check out let's encrypt.
But you really should not provide the private keys for your certificate in this project. 
Best regards
Marcel

**Comment by jpaulm** [comment]
There are a number of websites relating to this, e.g. https://stackoverflow.com/questions/7580508/getting-chrome-to-accept-self-signed-localhost-certificate  .  I guess the question is usability...  It sounds like I can support WS (after going back to an earlier version of my code), and maybe support WSS without localhost support, which might make testing my own code difficult...?  I will advertise for collaborators in a few weeks - probably on Slack or LinkedIn (or Twitter)...  Unless you would be able to spare a few cycles when I have some code ready...?  :-)  Thanks for your help anyway!

**Comment by marci4** [comment]
A self signed certificate will not be trusted by the browser by default. You have to do things like explicitly trusting the certificate or installing it to your certificate manager.

Sorry I don't have the time for this.

**Comment by jpaulm** [comment]
No problem, @marci4 !  You've been a big help already!

Thanks again, and stay safe!

**Comment by jpaulm** [comment]
Very quick question, @marci4 or @tootallnate : sorry to bother you, but I'm trying to figure out how to tell programmatically in WebSocketServer whether the client used ws: or wss: ...  

There is probably a simple solution, but I can't figure it out.   I thought WebSocket.getResourceDescriptor() would do it, but it doen't seem to contain that info...   Unless I'm using it on the wrong object...

Thanks in advance for any help you can give me!

**Comment by jpaulm** [comment]
Copied from my Slack post:  ... after struggling with this for several weeks, I found a solution that IMHO is rather neat!  The underlying software (Java-WebSocket) apparently cannot support both WS and WSS in the same run, so I have added an argument to the java command using the jar file, which in turns controls the contents of an IIP passed to the WebSocketReceive component.  Works quite nicely!  However, I have not been able to have the code consistently pick up a self-signed certificate using localhost (and in fact some experts say this can't be done), so my WSS option expects a Chrome client, with the chrome://flags/#allow-insecure-localhost option enabled.  For more detail see https://github.com/jpaulm/javafbp-websockets#readme . It would be extremely cool if someone could try the WSS (TLS) option using a valid certificate (and show me how to do so also)...  Anyone?!

Just adding your IDs, @marci4 @tootallnate , to let you know I am closing this...  Thanks!
